### PR TITLE
feat: ibc migrations

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

Closes: #1293 
Closes: #1413

* setups Interchain Accounts (daectivated by default)
* setups IBC Fees keeper

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -458,28 +480,87 @@ func New(
...
+	// SendPacket, since it is originating from the application to core IBC:
+	// icaAuthModuleKeeper.SendTx -> icaController.SendPacket -> fee.SendPacket -> channel.SendPacket
+
+	//// initialize ICA module with mock module as the authentication module on the controller side
+	//// TODO: https://github.com/umee-network/umee/issues/1413
+	// var icaControllerStack ibcporttypes.IBCModule
+	// icaControllerStack = ibcmock.NewIBCModule(&mockModule, ibcmock.NewIBCApp("", scopedICAMockKeeper))
+	// app.ICAAuthModule = icaControllerStack.(ibcmock.IBCModule)
+	// icaControllerStack = icacontroller.NewIBCMiddleware(icaControllerStack, app.ICAControllerKeeper)
+	// icaControllerStack = ibcfee.NewIBCMiddleware(icaControllerStack, app.IBCFeeKeeper)
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply app/app.go:542]: actually, I want to keep it because we will add it later on.

**Comment by robert-zaremba** [comment]
@adamewozniak ICA is disabled. We only properly set the IBC here and register the IBC modules.

**Comment by RafilxTenfen** [comment]
Still failing upgrade, testing...

**Comment by RafilxTenfen** [comment]
[90m1:30AM[0m [33mDBG[0m funded oracle [36mamount=[0m[] [36mmodule=[0mx/leverage
[90m1:30AM[0m [33mDBG[0m interest accrued [36mblock_height=[0m3045770 [36minterest=[0m [36mmodule=[0mx/leverage [36mreserved=[0m [36munix_time=[0m1663561800
[90m1:30AM[0m [32mINF[0m executed block [36mheight=[0m3045770 [36mmodule=[0mstate [36mnum_invalid_txs=[0m0 [36mnum_valid_txs=[0m0
[90m1:30AM[0m [1m[31mERR[0m[0m CONSENSUS FAILURE!!! [36merr=[0m"cannot delete latest saved version (10)" [36mmodule=[0mconsensus [36mstack=[0m"goroutine 217 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine.func2()\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:727 +0x4c\npanic({0x206c960, 0xc0f29c5f80})\n\t/usr/local/go/src/runtime/panic.go:884 +0x212\ngithub.com/cosmos/cosmos-sdk/store/rootmulti.(*Store).Commit(0xc000ed80a0)\n\t/home/rafilx/go/pkg/mod/github.com/umee-network/cosmos-sdk@v0.46.1-umee/store/rootmulti/store.go:433 +0x334\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Commit(0xc000ef2380)\n\t/home/rafilx/go/pkg/mod/github.com/umee-network/cosmos-sdk@v0.46.1-umee/baseapp/abci.go:322 +0x150\ngithub.com/tendermint/tendermint/abci/client.(*localClient).CommitSync(0xc01595e6c0)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/abci/client/local_client.go:264 +0xb6\ngithub.com/tendermint/tendermint/proxy.(*appConnConsensus).CommitSync(0xc0023344c0?)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/proxy/app_conn.go:93 +0x22\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).Commit(_, {{{0xb, 0x0}, {0xc010e2a390, 0x8}}, {0xc012a717a0, 0x16}, 0x2e796e, 0x2e798a, {{0xc0154ea400, ...}, ...}, ...}, ...)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/state/execution.go:228 +0x269\ngithub.com/tendermint/tendermint/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x0}, {0xc010e2a390, 0x8}}, {0xc012a717a0, 0x16}, 0x2e796e, 0x2e798a, {{0xc0154ea400, ...}, ...}, ...}, ...)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/state/execution.go:180 +0x6ee\ngithub.com/tendermint/tendermint/consensus.(*State).finalizeCommit(0xc08288d180, 0x2e798a)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:1659 +0xafd\ngithub.com/tendermint/tendermint/consensus.(*State).tryFinalizeCommit(0xc08288d180, 0x2e798a)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:1568 +0x2ff\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit.func1()\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:1503 +0x94\ngithub.com/tendermint/tendermint/consensus.(*State).enterCommit(0xc08288d180, 0x2e798a, 0x0)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:1541 +0xccf\ngithub.com/tendermint/tendermint/consensus.(*State).addVote(0xc08288d180, 0xc0a4f790e0, {0x0, 0x0})\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:2155 +0x18dc\ngithub.com/tendermint/tendermint/consensus.(*State).tryAddVote(0xc08288d180, 0xc0a4f790e0, {0x0?, 0x4818a6?})\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:1953 +0x2c\ngithub.com/tendermint/tendermint/consensus.(*State).handleMsg(0xc08288d180, {{0x2da9380?, 0xc0ab4b35d0?}, {0x0?, 0x0?}})\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:856 +0x170\ngithub.com/tendermint/tendermint/consensus.(*State).receiveRoutine(0xc08288d180, 0x0)\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:783 +0x505\ncreated by github.com/tendermint/tendermint/consensus.(*State).OnStart\n\t/home/rafilx/go/pkg/mod/github.com/tendermint/tendermint@v0.34.21/consensus/state.go:379 +0x12d\n"
[90m1:30AM[0m [32mINF[0m service stop [36mimpl=[0m{"Logger":{}} [36mmodule=[0mconsensus [36mmsg=[0m{} [36mwal=[0m/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal/wal
[90m1:30AM[0m [32mINF[0m service stop [36mimpl=[0m{"Dir":"/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal","Head":{"ID":"wboEunqqKzQ2:/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal/wal","Path":"/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal/wal"},"ID":"group:wboEunqqKzQ2:/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal/wal","Logger":{}} [36mmodule=[0mconsensus [36mmsg=[0m{} [36mwal=[0m/home/rafilx/projects/github.com/umee-network/umee/contrib/scripts/node-data/umeemain-local-testnet/node/data/cs.wal/wal
[90m1:30AM[0m [33mDBG[0m served RPC HTTP response [36mduration=[0m0 [36mmethod=[0mPOST [36mmodule=[0mrpc-server [36mremoteAddr=[0m127.0.0.1:43484 [36mstatus=[0m200 [36murl=[0m{"ForceQuery":false,"Fragment":"","Host":"","OmitHost":false,"Opaque":"","Path":"/","RawFragment":"","RawPath":"","RawQuery":"","Scheme":"","User":null}

Trying to debug

**Comment by RafilxTenfen** [comment]
we really need this for rc ?

**Review by RafilxTenfen — CHANGES_REQUESTED** [review]
Run ./contrib/scripts/umeemainnet_fork.sh  after make build and look at the logs, it will give an consensus failure, this does not happens on main

**Review by gsk967 — APPROVED** [review]
LGTM

**Review by toteki — APPROVED** [review]

**Comment by robert-zaremba** [comment]
converting into draft - we will continue this branch after v3.0 release

**Comment by robert-zaremba** [comment]
in fact closing it - will continue directly from commented code in https://github.com/umee-network/umee/pull/1425

**Pull request closed without merging**